### DISCUSSION TITLE
v0.0.8

### DISCUSSION THREAD

**Discussion by carson-katri** [carson-katri/dream-textures#374] [Announcements]
This update includes seamless upscaling, an improved render pass, a redesigned "Source Image" panel, model switching, DreamStudio integration, and more!

## Choose Your Installation
Several versions are available. Find the one that will work for you.

<details>
<summary>Windows</summary>

### NVIDIA GPU
* `dream_textures-windows-cuda.7z` - Local and cloud rendering with NVIDIA CUDA
    * Due to file size limits on GitHub, the add-on ZIP file is contained in a 7-Zip archive.
    * Install [7-Zip](https://www.7-zip.org/), right click on the downloaded `dream_textures-windows-cuda.7z` archive, select "7-Zip" > "Extract Here" to get the `dream_textures-windows-cuda.zip` file.
    * In Blender preferences open "Add-ons" > "Install...", then choose the `dream_textures-windows-cuda.zip` file.
* `dream_textures-windows-dreamstudio.zip` - Cloud rendering with DreamStudio

### AMD GPU
* `dream_textures-windows-dreamstudio.zip` - Cloud rendering with DreamStudio
> *Note* Local generation is not available for AMD GPUs.
</details>

<details>
<summary>macOS</summary>

### Apple Silicon (M1/M2)
* `dream_textures-macos-arm.zip` - Local and cloud rendering for Apple Silicon Macs
* `dream_textures-macos-arm-dreamstudio.zip` - Cloud rendering with DreamStudio

### Intel
* `dream_textures-macos-intel-dreamstudio.zip` - Cloud rendering with DreamStudio
> *Note* Local generation is not available for Intel Macs.
</details>

<details>
<summary>Linux</summary>

Installation must be completed manually. Follow the instructions under [Contributing](https://github.com/carson-katri/dream-textures/#contributing) for more details.
</details>

## Setup
After installing the appropriate add-on ZIP file, enable the add-on and expand it's preferences. There you will find further instructions for setup.

### Local Generation
To setup local generation, you will need a model checkpoint file, such as:
* [Stable Diffusion v1.4](https://huggingface.co/CompVis/stable-diffusion-v-1-4-original)
    * Download the file `sd-v1-4.ckpt`, and import it in the add-on preferences
* [Stable Diffusion v1.5](https://huggingface.co/runwayml/stable-diffusion-v1-5)
    * Download the file `v1-5-pruned-emaonly.ckpt`, and import it in the add-on preferences
* Fine-tuned model
    * Other fine-tuned models compatible with Stable Diffusion can be imported as well.

### DreamStudio Generation
To setup DreamStudio integration, find [your API key](https://beta.dreamstudio.ai/membership?tab=apiKeys) and enter it in the add-on preferences.

### Seamless Upscaling
Check "Seamless" to ensure tileable results from the AI upscaler.
<img width="381" alt="Screen Shot 2022-11-01 at 4 58 58 PM" src="https://user-images.githubusercontent.com/13581484/199339742-ce166534-1e0b-46cc-8689-2ba7f0a081b1.png">

### Render Pass Color Management
Color Management settings are now correctly applied when using the Dream Textures render pass. This change is automatically applied, and ensures results match between manual runs of Dream Textures and automated runs via the render pass.

### Advanced Presets
Choose defaults and create your own presets for advanced configuration.
<img width="452" alt="Screen Shot 2022-11-01 at 5 02 03 PM" src="https://user-images.githubusercontent.com/13581484/199340273-71396ec4-d43f-4a28-a74c-652d4219944f.png">

### File Batch and Iterations
Use a text file to run multiple prompts in a row, and set Iterations above 1 to try multiple seeds without interacting with the UI.
<img width="538" alt="Screen Shot 2022-11-01 at 5 04 36 PM" src="https://user-images.githubusercontent.com/13581484/199340709-594463e2-d6c1-4430-b00f-4974668e53d3.png">

### Init Image Redesign and Outpainting
The redesigned "Source Image" panel makes manipulating images easier to use. Various bugs regarding the "Mark Inpaint Area" brush have been resolved. Outpainting allows images to be extended beyond their original size. The "Prompt" mask source when inpainting allows you to choose a region to inpaint with a simple text prompt instead of a manual alpha-channel mask.
<img width="349" alt="Screen Shot 2022-11-01 at 5 07 58 PM" src="https://user-images.githubusercontent.com/13581484/199341813-9c6e58fd-3000-4bcf-b6ae-486b826a8cfa.png">

### Asymmetric Tiling
Tile along a single axis, or both for the original behavior. This can give interesting results for images such as infinitely scrolling backgrounds, decals, and more.
<img width="348" alt="Screen Shot 2022-11-01 at 5 09 07 PM" src="https://user-images.githubusercontent.com/13581484/199341988-1d23202a-cba7-486b-a717-1b41d3f7dd31.png">
<img width="182" alt="Screen Shot 2022-11-01 at 5 14 22 PM" src="https://user-images.githubusercontent.com/13581484/199342938-b0739d4e-ad61-4beb-abfc-eb959bbbf60c.png">

### [DreamStudio](https://beta.dreamstudio.ai) Integration
Connect your DreamStudio account to generate in the cloud. This is optional, and separate lighter builds with the DreamStudio backend only are provided for users with incompatible hardware.

### Model Switching
Import multiple models and switch between them quickly to experiment with fine-tuned and updated checkpoints.
<img width="359" alt="Screen Shot 2022-11-01 at 5 12 47 PM" src="https://user-images.githubusercontent.com/13581484/199342711-b928e1b5-75d5-41f2-86e1-44bc32bad441.png">

## Full Changelog
* Add documentation on various features by @carson-katri in https://github.com/carson-katri/dream-textures/pull/252
* Clear Modules from a Previous Install by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/261
* More Subprocess Refactoring and Cancel Generate Operator by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/274
* delete pointers before closing shared memory by @timmyL17 in https://github.com/carson-katri/dream-textures/pull/299
* Import Fixes by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/317
* Upscale Seamless Fix by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/318
* Apply OpenColorIO Transformations in Render Pass by @carson-katri in https://github.com/carson-katri/dream-textures/pull/319
* render pass set pixels fix by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/323
* Advanced Presets by @carson-katri in https://github.com/carson-katri/dream-textures/pull/324
* Check Weights Installed Fix by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/327
* Add File Batch and Re-Enable Iterations by @carson-katri in https://github.com/carson-katri/dream-textures/pull/332
* Fix InpaintAreaBrush behavior by @carson-katri in https://github.com/carson-katri/dream-textures/pull/333
* Init Image Redesign and Outpainting by @carson-katri in https://github.com/carson-katri/dream-textures/pull/334
* Add Asymmetric Tiling by @carson-katri in https://github.com/carson-katri/dream-textures/pull/335
* Add section linking to GH contributors by @carson-katri in https://github.com/carson-katri/dream-textures/pull/336
* DreamStudio integration by @carson-katri in https://github.com/carson-katri/dream-textures/pull/348
* Model switching by @carson-katri in https://github.com/carson-katri/dream-textures/pull/350
* tqdm decorator fix by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/369
* Render Pass Fix by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/370

## New Contributors
* @timmyL17 made their first contribution in https://github.com/carson-katri/dream-textures/pull/299

**Full Changelog**: https://github.com/carson-katri/dream-textures/compare/0.0.7...0.0.8

<hr /><em>This discussion was created from the release <a href='https://github.com/carson-katri/dream-textures/releases/tag/0.0.8'>v0.0.8</a>.</em>

**Comment by SolusGod** [comment]
Thank you for your hard work Carson!

**Comment by benrugg** [comment]
Really cool to see all these new features and enhancements. Love it.

**Comment by enzyme69** [comment]
Testing batch and iteration now.

I noticed right away that the History display of the batch is a bit strange. Lots of empty lines.
<img width="625" alt="Screenshot 2022-11-02 at 4 04 42 pm" src="https://user-images.githubusercontent.com/3952687/199402766-b82d62a4-0d22-4df2-9b7b-b5416356d3b7.png">

I wonder with styles you specify: custom, batch, texture, etc. I wonder if they should be a panel that can add "style presets" just to separate the prompt context and style.

Btw, this latest version crashes Blender, I might do reinstall now asap.

**Comment by enzyme69** [comment]
I tested "Dream Studio" online, apparently sometimes the generation is pretty fast, almost like a second to get the result. Sometimes it does wait a bit. It is not super clear on progress and result. I am sometimes being impatience and hitting the button a few times, maybe I am sending multiple tasks. 

Currently the result is named with just the SEED, which is probably okey, but could the name of files include the prompt text but truncated + the seed?

Furthermore, I feel a little bit more comfortable if I could see LOG text of how much processing time will take and already took.

**Comment by sjonsjine** [comment]
Would be handy for an option to include some Dream Textures settings (as prompt, steps, cfg scale, sampler...) in the rendered image. When experimenting/ tweaking settings this would be very helpfull.

**Comment by sjonsjine** [comment]
Using a certain same seed (not random) with size 512 x 512, gives a complete new rendering when using 1024 x 1024....that's strange isn't it?I tried the small size to find that image I want, because its very fast, use that seed with a different, bigger size but new image!

**Reply by carson-katri** [reply]
This is to be expected. Changing pretty much any parameter besides step count after a certain point will give a different result. To increase the size of an image use the built-in AI upscaling.

**Reply by sjonsjine** [reply]
Thank you for the info!

**Comment by JC-Sutter** [comment]
I don't understand the "inpaint: mask source - prompt". I always get the same result with this option. I do not raely understand what the "Confidence Treshhold" is

**Reply by carson-katri** [reply]
You write what you want to be masked out. So if you want to replace a mountain in the background you would enter “mountain”. The confidence threshold is how confident it has to be that a pixel is part of a mountain to mask it out. So if it’s set to 50%, it has to be at least 50% sure a pixel is part of a mountain to replace it.

**Reply by JC-Sutter** [reply]
Ah thank you for your answer. Do I need do make an alpha channel mask when I use "mask source - prompt" also?

**Reply by carson-katri** [reply]
No, the prompt is used to generate the mask.

**Reply by JC-Sutter** [reply]
Thank you. That works fine

Another question ist about the Outpaint. I always get a line or a hart crossing where at the expanded edge. I there a way to prevent that?

![image](https://user-images.githubusercontent.com/76898123/202269865-7bbf679b-8f54-47f7-b6cc-220126d8c38c.png)

**Reply by carson-katri** [reply]
Try increasing the blend option. Outpainting will improve in the next version using the v1.5 inpainting-specific model.

**Comment by JC-Sutter** [comment]
Is also possible to load the new StableDiffusion-v2?

https://huggingface.co/stabilityai/stable-diffusion-2-base

**Reply by carson-katri** [reply]
With #405 it will be

**Reply by JC-Sutter** [reply]
This means in the next version v0.0.9?

**Reply by carson-katri** [reply]
Yes

**Reply by Koningsbruggen** [reply]
Alright, noted. When is the 405 scheduled for release?

**Comment by sjonsjine** [comment]
ooooOOOOOOoooooooh!

Op di 6 dec. 2022 om 14:40 schreef Carson Katri ***@***.***>:

> Yes
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/carson-katri/dream-textures/discussions/374#discussioncomment-4323192>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFPWYYZ3LEBDO3UESBPXMLLWL463XANCNFSM6AAAAAARUOPDXM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
*Met vriendelijke groet/ with kind regards,*

*Freddy Peters - SjonSjine*

*M:+31(0)6 50 52 61 73*
*Veemkade 592*
*1019 BL Amsterdam*
*The Netherlands*
*https://www.sjonsjine.nl <https://www.sjonsjine.nl>*

https://linktr.ee/sjonsjine        - the whole package -

**Comment by Koningsbruggen** [comment]
Got the answer to below question: 
It is still in development. 

On [twitter ](https://twitter.com/CarsonKatri/status/1600248599254007810) I saw a feature that uses a depth map to generate the images. Where do I acces that? I installed the add on but see no 'depth' option to select in the advanced section. 

![image](https://user-images.githubusercontent.com/116488392/206227643-f188c5b3-3a4f-408d-a3af-c75041a1fc31.png)

**Reply by carson-katri** [reply]
The PR is still open #409

**Reply by Koningsbruggen** [reply]
Oh cool, I would love to try it! Would just have to find out how to do that from the page you shared.

**Reply by Koningsbruggen** [reply]
Oh wait I misunderstood, PR is still open means pull request is still open meaning you are open for feedback and still building on it but it does not mean open for beta testers. I had to use ChatGPT to understand the lingo. I really enjoy using dream textures. Using dept maps would truly be a game changer. Especially if you could combine it with other models, like use the depth model to make a mask and another model to turn the generation into a certain style. That would be super helpful to make a scene more coherent and would also allow people to use more open sources blender objects, as you could just use the base model and retexture it in the style of your scene. Thanks for your work!

**Reply by Koningsbruggen** [reply]
Ha cool, I could sponsor you, great feature, never used Github before

**Reply by Koningsbruggen** [reply]
The 2+ SD models don't currently work, correct?

**Reply by jclevet** [reply]
> The 2+ SD models don't currently work, correct?

We have to wait v0.0.9 for this feature...
